src/rules.ts

```diff
@@ -435,6 +438,18 @@ export class RuleChain extends LogicRule {
   }
 }
 
+export class RuleRaceChain extends RuleChain {
```

Comment:
very nice! perhaps better rename it to `RuleRace` to differentiate the two logic constructors.